components/TARGET_PSA/inc/psa/service.h

```diff
@@ -17,8 +17,14 @@
 
 #if defined(TARGET_TFM)
 #include "interface/include/psa_service.h"
-#else
+#define SPM_PANIC(format, ...) \
+{ \
+    while(1){}; \
+}
+#elif defined(TARGET_MBED_SPM)
 #include "TARGET_MBED_SPM/psa_defs.h"
 #include "TARGET_MBED_SPM/COMPONENT_SPE/spm_server.h"
 #include "TARGET_MBED_SPM/COMPONENT_SPE/spm_panic.h"
+#else
+#error "Compiling psa service header on non-secure target is not allowed"
```

Comment:
note that the error is misleading.
TARGET_MBED_SPM is applied to both secure and nonsecure targets

Reply:
Maybe `#error "Compiling psa service header on a PSA emulated target is not allowed"`

Reply:
i would phrase it as "expected"